src/ZoneServer/Skills/Handlers/Highlander/SkyLiner.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.L10N;
+using Melia.Shared.Tos.Const;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Highlander
+{
+	/// <summary>
+	/// Handler for the Highlander skill SkyLiner.
+	/// </summary>
+	[SkillHandler(SkillId.Highlander_SkyLiner)]
+	public class SkyLiner : IGroundSkillHandler, IDynamicCasted
+	{
+		private const int TotalHits = 4;
+
+		public void StartDynamicCast(Skill skill, ICombatEntity caster)
+		{
+			if (caster is Character character)
```

Comment:
Based on the name of this variable and how you're setting it, it seems like it might be better suited for a general purpose variable that would be set from the packet handlers. Is this a Sky Liner specific mechanic? If so, the variable name should reflect that. If not, we should probably look into establishing a clearer way to use it.

Reply:
It's not skyliner specific, I think every dynamic cast skill would use this.  I don't quite know where you're suggesting you want to add it, we can probably discuss this more in the development chat.

Reply:
If this is a general function of dynamically casted skills, we don't want to repeat this in every single dynamic skill handler. Instead, the packet handlers could set the variable and the skill handler could run a proper loop while the variable is set. Ideally this variable would also be an actual property, rather than a character variable, to "legitimize" it and make it easier and clearer what it is. And seeing how it would be a core part of combat, it should most likely be part of ICombatEntity.

Reply:
I've added it to the CombatComponent, similar to the AttackState.  I specifically tested to see if the client would send the "endDynamicCasting" packet even if you hold it to the maximum length (at which point it ends automatically) and it still does, so setting the casting state to false in endDynamicCasting is safe.